e2e/specs/st_set_page_config_icon.spec.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright 2018-2022 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe("st.set_page_config", () => {
+  before(() => {
+    cy.loadApp("http://localhost:3000/");
+  });
+
+  it("sets the page favicon with ico file", () => {
+    cy.get("link[rel='shortcut icon']")
+      .should("have.attr", "href")
+      .should("contain", "png");
```

Comment:
This would produce a false negative. The default is a PNG if not specified, so if `assets/favicon.ico` is not found. it will default to a png 🤣

![7EDF8D93-D164-40E5-84D5-29C6FBD40A4B](https://user-images.githubusercontent.com/69432/152255654-c47189e1-13ab-42ad-b15d-a1a9313c08cb.png)